packages/packages/store/src/__tests__/store.test.tsx

```diff
@@ -0,0 +1,198 @@
+	);
+
+	return {
+		slice,
+		store,
+		wrapper,
+	};
+};
+
+describe( '@elementor/store', () => {
+	it( 'should verify the initial state of the slice', () => {
+		const storeService = createStoreService();
+
+		const { wrapper } = createStoreEntities( storeService );
+
+		const { result } = renderHook( () => useSelector( ( state: SliceStateRoot ) => state.slice.value ), { wrapper } );
+
+		expect( result.current ).toBe( 1 );
+	} );
+
+	it( 'should update the state value of the slice', () => {
+		const storeService = createStoreService();
+
+		const { slice, wrapper } = createStoreEntities( storeService );
+
+		const { result } = renderHook( () => {
+			const dispatchAction = useDispatch();
+
+			dispatchAction( slice.actions.setValue( 3 ) );
+
+			return useSelector( ( state: SliceStateRoot ) => state.slice.value );
+		}, { wrapper } );
+
+		expect( result.current ).toBe( 3 );
+	} );
+
+	it( 'should throw an error when trying to register a slice with the same name more than once', () => {
+		const storeService = createStoreService();
+
+		storeService.registerSlice( {
+			name: 'slice',
+			initialState: {
+				value: 1,
+			},
+			reducers: {},
+		} );
+
+		expect( () => {
+			storeService.registerSlice( {
+				name: 'slice',
+				initialState: {
+					value: 1,
+				},
+				reducers: {},
+			} );
+		} ).toThrowError();
+	} );
+
+	it( 'should throw an error when trying to re-create the store on the same storeService instance', () => {
+		const storeService = createStoreService();
+
+		storeService.registerSlice( {
+			name: 'slice',
+			initialState: {
+				value: 1,
+			},
+			reducers: {},
+		} );
+
+		storeService.createStore();
+
+		expect( () => storeService.createStore() ).toThrowError();
+	} );
+
+	it( 'should register a middleware that blocks the state update by not running next(action)', () => {
+		const storeService = createStoreService();
+
+		storeService.registerMiddleware( () => () => () => {} );
+
+		const { slice, wrapper } = createStoreEntities( storeService );
+
+		const { result } = renderHook( () => {
+			const dispatchAction = useDispatch();
+
+			dispatchAction( slice.actions.setValue( 4 ) );
+
+			return useSelector( ( state: SliceStateRoot ) => state.slice.value );
+		}, { wrapper } );
+
+		expect( result.current ).toBe( 1 );
+	} );
+
+	it( 'should register a middleware that does not intefer with the state update', () => {
+		const storeService = createStoreService();
+
+		storeService.registerMiddleware( () => ( next: Dispatch<AnyAction> ) => ( action: any ) => {
+			next( action );
+		} );
+
+		const { slice, wrapper } = createStoreEntities( storeService );
+
+		const { result } = renderHook( () => {
+			const dispatchAction = useDispatch();
+
+			dispatchAction( slice.actions.setValue( 4 ) );
+
+			return useSelector( ( state: SliceStateRoot ) => state.slice.value );
+		}, { wrapper } );
+
+		expect( result.current ).toBe( 4 );
+	} );
+
+	it( 'should dispatch an action without using the useDispatch hook', () => {
```

Comment:
if you're testing the dispatch, maybe also the without the selector (could be in the same test, just remove the `useSelector` and use `store.getState()['slice'].value` or something)

Reply:
When I remove the renderHook, and then trigger a dispatch, the state value (store.getState()..) does not change synchronously..